pkg/types/baremetal/validation/platform.go

```diff
@@ -84,6 +108,16 @@ func ValidatePlatform(p *baremetal.Platform, n *types.Networking, fldPath *field
 	if err := validateIPNotinMachineCIDR(p.BootstrapProvisioningIP, n); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("bootstrapHostIP"), p.BootstrapProvisioningIP, err.Error()))
 	}
+	if p.BootstrapOSImage != "" {
+		if err := validateOSImageURI(p.BootstrapOSImage, "BootstrapOSImage"); err != nil {
```

Comment:
why are we passing `BootstrapOSImage` to this function.. 
looks like for error: https://github.com/openshift/installer/pull/2757/files#diff-144e602ba4c4174e8b2e21f6d43d59d6R40

I don't think it's correct.

https://github.com/openshift/installer/pull/2757/files#diff-144e602ba4c4174e8b2e21f6d43d59d6R113
`field.Invalid` already has the field Name in the error returned..?

Reply:
Yeah that was not . This was a last minute change. Could have just used the URI, which is what I've done in the latest commit which also uses regexp to check expected error.

Also regarding your second observation, I'm just using the same style used in the rest of the validation code.